src/dxgi.cpp

```diff
@@ -496,28 +496,63 @@ namespace bgfx
 			DX_RELEASE_I(factory5);
 		}
 
-		DXGI_SWAP_CHAIN_DESC scd;
-		scd.BufferDesc.Width  = _scd.width;
-		scd.BufferDesc.Height = _scd.height;
-		scd.BufferDesc.RefreshRate.Numerator   = 1;
-		scd.BufferDesc.RefreshRate.Denominator = 60;
-		scd.BufferDesc.Format = _scd.format;
-		scd.BufferDesc.ScanlineOrdering = DXGI_MODE_SCANLINE_ORDER_UNSPECIFIED;
-		scd.BufferDesc.Scaling = DXGI_MODE_SCALING_UNSPECIFIED;
-		scd.SampleDesc.Count   = 1;
-		scd.SampleDesc.Quality = 0;
-		scd.BufferUsage  = _scd.bufferUsage;
-		scd.BufferCount  = _scd.bufferCount;
-		scd.OutputWindow = (HWND)_scd.nwh;
-		scd.Windowed     = _scd.windowed;
-		scd.SwapEffect   = _scd.swapEffect;
-		scd.Flags        = scdFlags;
-
-		hr = m_factory->CreateSwapChain(
-			  _device
-			, &scd
-			, reinterpret_cast<IDXGISwapChain**>(_swapChain)
+		IDXGIFactory2* factory2;
+		hr = m_factory->QueryInterface(IID_IDXGIFactory2, (void**)&factory2);
+		if (SUCCEEDED(hr)) {
```

Comment:
Fix style...

Reply:
Also this seems pointless, since Windows is using IDXGIFactory5 anyway. You can always use DESC1

Reply:
Fixing style.
I thought the above factory5 lines just check for those features but doesn't require IDXGIFactory5.

Reply:
I'm talking about this: https://github.com/bkaradzic/bgfx/blob/f534946822b13b89721969a4281bb1af93c53af9/src/dxgi.h#L61

`m_factory` is already `IDXGIFactory5`

Reply:
So just remove `if / else` stuff, and use `DXGI_SWAP_CHAIN_DESC1` without checks.

Reply:
Alright I remove the check. I also put it outside the #if because the #else below has the same DXGI_SWAP_CHAIN_DESC1 setup.